### DISCUSSION TITLE
Scoping custom Element Actions by specific site/sites

### DISCUSSION THREAD

**Discussion by mrog-pbs** [craftcms/cms#15159] [Q&A] [answered]
`EVENT_REGISTER_ACTIONS` is triggered in the static context, and `RegisterElementActionsEvent` only carries the _key_ (string) of the element source from which the action is invoked, _not_ the source _site_.

I'd like to conditionally limit a custom action to only be available in certain source sites.

Is there any way to sniff the active site for an element source for the purpose of registering site-specific element actions?

(I initially thought to use `Craft::$app->sites->currentSite`, but actually that resolves to the site _from which the CP is being viewed_ — i.e. the primary site — _NOT_ the source site of a particular element source being used.)

**Comment by brandonkelly** [comment, marked as answer]
`currentSite` will always be the primary site for control panel requests. You can get the site that the element index is loading via `craft\helpers\Cp::requestedSite()`.

**Reply by mrog-pbs** [reply]
Perfect — thank you! 🙏🏼
